src/state-manager/TransactionConsensus.ts

```diff
@@ -1024,6 +1094,565 @@ class TransactionConsenus {
+          queueEntry.hasSentFinalReceipt = true
+          Comms.sendGossip(
+            'poqo-receipt-gossip',
+            payload,
+            null,
+            null,
+            queueEntry.transactionGroup,
+            false,
+            4,
+            payload.txid,
+            '',
+            true
+          )
+          this.stateManager.transactionQueue.factTellCorrespondingNodesFinalData(queueEntry)
+        } finally {
+          profilerInstance.scopedProfileSectionEnd('poqo-send-receipt')
+        }
+      }
+    )
+
+    const poqoSendReceiptBinary: Route<InternalBinaryHandler<Buffer>> = {
+      name: InternalRouteEnum.binary_poqo_send_receipt,
+      handler: async (payload, respond, header) => {
+        const route = InternalRouteEnum.binary_poqo_send_receipt
+        this.profiler.scopedProfileSectionStart(route)
+        nestedCountersInstance.countEvent('internal', route)
+        profilerInstance.scopedProfileSectionStart(route, false, payload.length)
+
+        const errorHandler = (
+          errorType: RequestErrorEnum,
+          opts?: { customErrorLog?: string; customCounterSuffix?: string }
+        ): void => requestErrorHandler(route, errorType, header, opts)
+
+        try {
+          const requestStream = getStreamWithTypeCheck(payload, TypeIdentifierEnum.cPoqoSendReceiptReq)
+          if (!requestStream) {
+            return errorHandler(RequestErrorEnum.InvalidRequest)
+          }
+
+          const readableReq = deserializePoqoSendReceiptReq(requestStream)
+
+          const queueEntry = this.stateManager.transactionQueue.getQueueEntrySafe(readableReq.txid)
+          if (queueEntry == null) {
+            /* prettier-ignore */ nestedCountersInstance.countEvent('poqo', 'binary/poqo_send_receipt: no queue entry found')
+            return
+          }
+
+          if (queueEntry.poqoReceipt) {
+            // We've already handled this
+            return
+          }
+
+          const executionGroupNodes = new Set(queueEntry.executionGroup.map((node) => node.publicKey))
+          const hasTwoThirdsMajority = this.verifyAppliedReceipt(readableReq, executionGroupNodes)
+          if (!hasTwoThirdsMajority) {
+            /* prettier-ignore */ if (logFlags.error) this.mainLogger.error(`Receipt does not have the required majority for txid: ${readableReq.txid}`)
+            nestedCountersInstance.countEvent('poqo', 'poqo-send-receipt: Rejecting receipt because no majority')
+            return
+          }
+
+          if (logFlags.verbose)
+            this.mainLogger.debug(
+              `POQo: Received receipt from aggregator for ${queueEntry.logID} starting CT2 for data & receipt`
+            )
+          const receivedReceipt = readableReq as AppliedReceipt2
+          queueEntry.poqoReceipt = receivedReceipt
+          queueEntry.appliedReceipt2 = receivedReceipt
+          queueEntry.recievedAppliedReceipt2 = receivedReceipt
+          queueEntry.hasSentFinalReceipt = true
```

Comment:
if gossip will fail or corresponding tell, this true value will not represent fact that it was sent

Reply:
note the non-binary endpoint with this will be removed.

I think in general our systems to not track if gossip out fails.  I know you have mentioned the good point before in meetings that since it is tcp will know if transport fails on us.    I am adding this general topic an   ARCH review task

ARCH-9